lib/v2/telegram/channel.js

```diff
@@ -3,29 +3,111 @@ const cheerio = require('cheerio');
+    // VIDEO_STICKER: ['[Video Sticker]', '[Video Sticker]'],  // not supported yet by t.me
+    UNSUPPORTED: ['[Unsupported]', '🚫'],
+};
 
 module.exports = async (ctx) => {
     const username = ctx.params.username;
-    const searchQuery = ctx.params.searchQuery;
+    let routeParams = ctx.params.routeParams;
+    let showLinkPreview = true;
+    let showViaBot = true;
+    let showReplyTo = true;
+    let showFwdFrom = true;
+    let showFwdFromAuthor = true;
+    let showInlineButtons = false;
+    let showMediaTagInTitle = true;
+    let showMediaTagAsEmoji = true;
+    let includeFwd = true;
+    let includeReply = true;
+    let includeServiceMsg = true;
+    let includeUnsupportedMsg = false;
+    let searchQuery = routeParams; // for backward compatibility
+    if (routeParams && routeParams.search(/(^|&)(show(LinkPreview|ViaBot|ReplyTo|FwdFrom(Author)?|InlineButtons|MediaTag(InTitle|AsEmoji))|include(Fwd|Reply|(Service|Unsupported)Msg)|searchQuery)=/) !== -1) {
+        routeParams = querystring.parse(ctx.params.routeParams);
+        showLinkPreview = !!fallback(undefined, queryToBoolean(routeParams.showLinkPreview), showLinkPreview);
+        showViaBot = !!fallback(undefined, queryToBoolean(routeParams.showReplyTo), showViaBot);
+        showReplyTo = !!fallback(undefined, queryToBoolean(routeParams.showViaBot), showReplyTo);
+        showFwdFrom = !!fallback(undefined, queryToBoolean(routeParams.showFwdFrom), showFwdFrom);
+        showFwdFromAuthor = !!fallback(undefined, queryToBoolean(routeParams.showFwdFromAuthor), showFwdFromAuthor);
+        showInlineButtons = !!fallback(undefined, queryToBoolean(routeParams.showInlineButtons), showInlineButtons);
+        showMediaTagInTitle = !!fallback(undefined, queryToBoolean(routeParams.showMediaTagInTitle), showMediaTagInTitle);
+        showMediaTagAsEmoji = !!fallback(undefined, queryToBoolean(routeParams.showMediaTagAsEmoji), showMediaTagAsEmoji);
+        includeFwd = !!fallback(undefined, queryToBoolean(routeParams.includeFwd), includeFwd);
+        includeReply = !!fallback(undefined, queryToBoolean(routeParams.includeReply), includeReply);
+        includeServiceMsg = !!fallback(undefined, queryToBoolean(routeParams.includeServiceMsg), includeServiceMsg);
+        includeUnsupportedMsg = !!fallback(undefined, queryToBoolean(routeParams.includeUnsupportedMsg), includeUnsupportedMsg);
+        searchQuery = fallback(undefined, routeParams.searchQuery, null);
+    }
+
     const resourceUrl = searchQuery ? `https://t.me/s/${username}?q=${encodeURIComponent(searchQuery)}` : `https://t.me/s/${username}`;
 
-    const { data } = await got.get(resourceUrl);
+    const data = await ctx.cache.tryGet(
+        resourceUrl,
+        async () => {
+            const response = await got.get(resourceUrl);
+            return response.data;
+        },
+        300
+    ); // cache it for 5 minutes to avoid reload when routeParams changes
```

Comment:
Can I safely use `config.cache.routeExpire` here? Since those who have their own instance may set a route expire time shorter than 5min.

Reply:
If so, then those who want more frequent update on the content listing cannot benefit from it.
On the other hand, some instances(e.g., demo instance) may have a longer `routeExpire` value to prevent being throttled may also get into trouble.

Reply:
> If so, then those who want more frequent update on the content listing cannot benefit from it.

I don't quite understand. It is `routeExpire` instead of `contentExpire`. If `routeParams` is unchanged, this cache never comes in handy because of the route cache hit. Only if `routeParams` changed, this cache can come in handy. 
Even if I don't apply a cache here, the problem is that, is there anyone who needs to benefit from changing `routeParams` to bypass the route cache?

> On the other hand, some instances(e.g., demo instance) may have a longer `routeExpire` value to prevent being throttled may also get into trouble.

Could you give an example of how this might happen? If `routeExpire` is more than 5min, then only cache it for 5min, would it be better?
My main concern is that if a lot of users subscribed to the same channel but applies different `routeParams`, it can cause a lot unnecessary duplicated web requests if cache not applied.

Reply:
> It is `routeExpire` instead of `contentExpire`. If `routeParams` is unchanged, this cache never comes in handy because of the route cache hit.

If one set `config.cache.routeExpire` to 0 (completely disable route cache), with this settings, new feed items will delayed by 5 mins which does not seem alot. But I guess they set to 0 for instant notification and they have their reasons. Adding a fixed limit to it is not a good thing.

> Could you give an example of how this might happen?

<https://rsshub.app> has a `routeExpire` of 120mins.

> My main concern is that if a lot of users subscribed to the same channel but applied different routeParams, it can cause a lot of unnecessary duplicated web requests if cache is not applied.

I see your pov. `ctx.cache.tryGet()` already provide a cache time of 5 mins for key `resourceUrl` if no cache time provided and changing `routeParams` will not invalid caching according to your code.

> those who have their own instance may set a route expire time shorter than 5min.

If they set it shorter than 5 mins or even 0, they know the risks(generate lots of unnecessary traffics and getting ip banned) and benefits(quick update) of it.

Reply:
> If one set `config.cache.routeExpire` to 0 (completely disable route cache), with this settings, new feed items will delayed by 5 mins which does not seem alot. But I guess they set to 0 for instant notification and they have their reasons. Adding a fixed limit to it is not a good thing.

Yeah, so I've asked if using `routeExpire` (instead of `contentExpire` by default, or fixed 5min) is a good idea.

> `ctx.cache.tryGet()` already provide a cache time of 5 mins for key `resourceUrl` if no cache time provided

5 mins? Doesn't it default to be `contentExpire`? The reason why I insist that here should not use default is that `contentExpire` is usually much too long. Either no caching is used, or use `routeExpire` as cache time. Using `contentExpire` as cache time is unbearable.

```js
tryGet: async (key, getValueFunc, maxAge = config.cache.contentExpire) => {
```

<hr>

Now I know what the problem really is. If I use `ctx.cache.tryGet()`, RSSHub will enforce refreshing the expiration time on each read. And this refresh will make provided `maxAge` useless — RSSHub always reset the expiration time to `config.cache.routeExpire`!
There is a workaround to make content cache behave like route cache — use `ctx.cache.get({refresh: false})` and `ctx.cache.set()` instead. What's your opinion about that?

I also wonder why `maxAge` is ineffective when refreshing, and why enforcing refresh on `ctx.cache.tryGet()`. If refreshing is selectable, those routes with tons of formatting options (e.g. Weibo, Twitter) can cache the initial entry too, just like route caching, making anti-rate-limit and web traffic saving more effective.

Reply:
> There is a workaround to make content cache behave like route cache — `use ctx.cache.get({refresh: false})` and `ctx.cache.set()` instead. What's your opinion about that?

Sure, you can use it to cache the source page(with `refresh: false` on both set/get in this case).

> why enforcing refresh on `ctx.cache.tryGet()`.

Related https://github.com/DIYgod/RSSHub/issues/1182